File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/RxAutoCompleteTextView.java
package com.jakewharton.rxbinding2.widget;

import androidx.annotation.CheckResult;
import androidx.annotation.NonNull;
import android.widget.AutoCompleteTextView;
import io.reactivex.Observable;
import io.reactivex.functions.Consumer;

import static com.jakewharton.rxbinding2.internal.Preconditions.checkNotNull;

/**
 * Static factory methods for creating {@linkplain Observable observables} and {@linkplain Consumer
 * actions} for {@link AutoCompleteTextView}.
 */
public final class RxAutoCompleteTextView {
  /**
   * Create an observable of item click events on {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   */
  @CheckResult @NonNull
  public static Observable<AdapterViewItemClickEvent> itemClickEvents(
      @NonNull AutoCompleteTextView view) {
    checkNotNull(view, "view == null");
    return new AutoCompleteTextViewItemClickEventObservable(view);
  }

  private RxAutoCompleteTextView() {
    throw new AssertionError("No instances.");
  }
}

File: rxbinding-kotlin/src/main/kotlin/com/jakewharton/rxbinding2/widget/RxAutoCompleteTextView.kt
package com.jakewharton.rxbinding2.widget

import android.widget.AutoCompleteTextView
import io.reactivex.Observable
import io.reactivex.functions.Consumer

/**
 * Create an observable of item click events on `view`.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
inline fun AutoCompleteTextView.itemClickEvents(): Observable<AdapterViewItemClickEvent> = RxAutoCompleteTextView.itemClickEvents(this)

/**
 * An action that sets the optional hint text that is displayed at the bottom of the the matching
 * list. This can be used as a cue to the user on how to best use the list, or to provide extra
 * information.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
inline fun AutoCompleteTextView.completionHint(): Consumer<in CharSequence> = RxAutoCompleteTextView.completionHint(this)

/**
 * An action that specifies the minimum number of characters the user has to type in the edit box
 * before the drop down list is shown. When threshold is less than or equals 0, a threshold of 1
 * is applied.
 *
 * *Warning:* The created observable keeps a strong reference to `view`. Unsubscribe
 * to free this reference.
 */
inline fun AutoCompleteTextView.threshold(): Consumer<in Int> = RxAutoCompleteTextView.threshold(this)
